File: ejercicios hechos/Python/estructuras alternativas/act20.py
#Programa: act20.py
#Propósito: una compañía de transporte internacional tiene servicio en algunos países de América del Norte, América Central, América del Sur, Europa y Asia. El costo por el servicio de transporte se basa en el peso del paquete y la zona a la que va dirigido.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 17/10/2019
#
#Variables a usar:
# zona y peso son los valores del paquete
# precio será el resultado
#
#Algoritmo:
# lEER zona y peso
# si el peso es mayo a 5kg se rechaza el pedido
# si la zona es America del Norte 24 euros kg
# si la zona es America Central 20 euros kg
# si la zona es America del Sur 21 euros kg
# si la zona es Europa 10 euros kg
# si la zona es Asia 18 euros kg
# calculamos y mostramos en pantalla

print("calcular el precio de un paquete")
print("---------------------\n")

# Leemos los datos
zona = str(input("Introduce la zona a la que desea enviar: "))
peso = float(input("Introduce el peso del paquete, recueda no debe pesar más de 5kg: "))

# Realizamos las comprobaciones y calculos
if peso > 5:
    print("Lo sentimos pero no podemos enviar pesos superiores a 5kg")

else:
    if zona == "América del Norte":
        precio = round((peso*1000) * 24,2)
        print(f"El precio es {precio} euros")
    else:
        if zona == "América del Sur":
            precio = round((peso*1000) * 21,2)
            print(f"El precio es {precio} euros")
        
        else:
            if zona == "América Central":
                precio = round((peso*1000) * 20,2)
                print(f"El precio es {precio} euros")
        
            else:
                if zona == "Europa":
                    precio = round((peso*1000) * 10,2)
                    print(f"El precio es {precio} euros")
        
                else:
                    if zona == "Asia":
                        precio = round((peso*1000) * 18,2)
                        print(f"El precio es {precio} euros")

File: ejercicios hechos/Java/estructuras alternativas/act20.java
/*Programa: act20.java
 * 
 *Propósito: Una compañía de transporte internacional tiene servicio en algunos países de América del Norte, América Central, América del Sur, Europa y Asia. El costo por el servicio de transporte se basa en el peso del paquete y la zona a la que va dirigido.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 10/11/2019
 *
 *Análisis:
 * El coste del servicio se basa en el peso y a donde se dirige
 * Para paquetes mayores a 5KG no hay entregas
 * 
 *Diseño:
 * 1. Leer peso
 * 2. Si no es correcto >0 y <5000 mostrar mensaje de error
 * 3. Si es correcto:
 * 4. Leer zona
 * 5. Según el valor de la zona calculo el coste del transporte.
 * 6. Mostrar el coste de transporte.
 * 
*/
import java.util.Scanner;
 
public class act20 {
    public static void main(String[] args) {
        System.out.println("Programa para calcular el precio de un transporte");
        System.out.println("-------------------------------------------------");
        Scanner s = new Scanner(System.in);
        int peso;
        int zona;

        System.out.print("¿Qué peso tiene el paquete (en gramos)?: ");
        peso = s.nextInt();

        if (peso>0 && peso<=5000) {
          System.out.println("1.- América del Norte");
          System.out.println("2.- América Central");
          System.out.println("3.- América del Sur");
          System.out.println("4.- Europa");
          System.out.println("5.- Asia");
          System.out.print("A que zona se reparte (1-5): ");
          zona = s.nextInt();
          switch (zona) {
            case 1:
              System.out.println("Coste: "+peso*24+" euros.");
              break;
            case 2:
              System.out.println("Coste: "+peso*20+" euros.");
              break;
            case 3:
              System.out.println("Coste: "+peso*21+" euros.");
              break;
            case 4:
              System.out.println("Coste: "+peso*10+" euros.");
              break;
            case 5:
              System.out.println("Coste: "+peso*18+" euros.");
              break;
            default:
              System.out.println("Zona incorrecta.");
          }
        } else {
          System.out.println("Peso incorrecto (no podemos transportar paquetes de más de 5Kg).");
        }
    }
}
